nix/build-dapp-package.nix

```diff
@@ -0,0 +1,88 @@
+{ pkgs }:
+
+let
+  remappings = xs:
+    builtins.foldl' pkgs.lib.mergeAttrs {}
+      (builtins.map
+        (x: {
+          "${x.name}/" = "${x}/dapp/${x.name}/src/";
+         } // x.remappings)
+         xs);
+  libPaths = xs:
+    builtins.foldl' pkgs.lib.mergeAttrs {}
+      (builtins.map
+        (x: {
+          "${x.name}" = "${x}/dapp/${x.name}/src";
+         } // x.libPaths)
+         xs);
+in
+  {
+    name
+  , src
+  , deps ? []
+  , solc ? pkgs.solc
+  , test-hevm ? pkgs.dapp2.test-hevm
+  , solcFlags ? ""
+  , doCheck ? true
+  , extract ? false
+  , ... }@args:
+    pkgs.stdenv.mkDerivation ( rec {
+      inherit doCheck extract;
+
+      buildInputs = [ pkgs.jq solc test-hevm ];
+      passthru = {
+        remappings = remappings deps;
+        libPaths = libPaths deps;
+      };
+
+      REMAPPINGS =
+        pkgs.lib.mapAttrsToList
+          (k: v: k + "=" + v)
+          passthru.remappings;
+
+      LIBSCRIPT =
+        pkgs.lib.mapAttrsToList
+          (k: v: ''
+            ln -s ${v} lib/${k}
+          '')
+          passthru.libPaths;
+
+      buildPhase = args.buildPhase or ''
+        opts=(--combined-json=abi,bin,bin-runtime,srcmap,srcmap-runtime,ast,metadata --overwrite)
+
+        mkdir -p out
+
+        mapfile -t files < <(find "$src" -name '*.sol')
+        json_file="out/dapp.sol.json"
+        (set -x; ${solc}/bin/solc $REMAPPINGS "''${opts[@]}" $solcFlags "''${files[@]}" > "$json_file")
+
+        mkdir -p lib
+        source <(echo "$LIBSCRIPT")
+
+        if [ "$extract" == 1 ]; then
+          mapfile -t contracts < <(<"$json_file" jq '.contracts|keys[]' -r | sort -u -t: -k2 | sort)
+          data=$(<"$json_file" jq '.contracts' -r)
+          total=''${#contracts[@]}
+          echo "Extracting build data... [Total: $total]"
+          for path in "''${contracts[@]}"; do
+            fileName="''${path#*:}"
+            contract=$(echo "$data" | jq '.["'"$path"'"]')
+            echo "$contract" | jq '.["abi"]' -r > "out/$fileName.abi"
+            echo "$contract" | jq '.["bin"]' -r > "out/$fileName.bin"
+            echo "$contract" | jq '.["bin-runtime"]' -r > "out/$fileName.bin-runtime"
+            echo "$contract" | jq '.["metadata"]' -r > "out/$fileName.metadata"
+          done
+        fi
+      '';
+
+      checkPhase = args.checkPhase or ''
+        DAPP_OUT=out dapp2-test-hevm
```

Comment:
out is the default value for `DAPP_OUT` FYI

Reply:
Where do you see that? The default would have to be set [here](
https://github.com/dapphub/dapptools/tree/15c241a82183da46f5e170350dd378c99e1f264b/nix/dapp/dapp-test-hevm.nix#L5) for it to take effect.

Reply:
dapp2-test-hevm is not in `libexec/`, meaning the `libexec/dapp` script is not run, so variables exported there have no effect.

Reply:
Im wrobg

Reply:
What??

Reply:
I retract my previous statement